avalon/nuke/lib.py

```diff
@@ -3,82 +3,283 @@
+                if "__divider__" in k["name"]:
+                    knob = nuke.Text_Knob("")
+                    node.addKnob(knob)
+                elif k["name"] not in node.knobs().keys():
+                    knob = eval("nuke.{type}('{name}')".format(**k))
+                    node.addKnob(knob)
+                    try:
+                        knob.setValue(k['value'])
+                    except TypeError as E:
+                        print(E)
+            else:
+                if k["name"] not in node.knobs().keys():
+                    knob = eval((
+                        "nuke.{type}('{name}',"
+                        "'{value}', {group})"
+                    ).format(**k))
+                    node.addKnob(knob)
+
+        # add avalon knobs for imprinting data
+        for k, v in data.items():
+            label = k
+            name = prefix + label
+            value = str(v)
+            if label in non_hiden:
+                if name not in node.knobs().keys():
+                    log.info("Setting: `{0}` to `{1}`".format(name, value))
+                    knob = eval((
+                        "nuke.String_Knob('{name}','{label}',"
+                        "'{value}')"
+                    ).format(
+                        name=name,
+                        label=label,
+                        value=value
+                    ))
+                    node.addKnob(knob)
+                else:
+                    log.info("Updating: `{0}` to `{1}`".format(name, value))
+                    node[name].setValue(value)
+            else:
+                if name not in node.knobs().keys():
+                    log.info("Setting: `{0}` to `{1}`".format(name, value))
+                    knob = eval((
+                        "nuke.Text_Knob('{name}','{label}',"
+                        "'{value}')"
+                    ).format(
+                        name=name,
+                        label=label,
+                        value=value
+                    ))
+                    node.addKnob(knob)
+                else:
+                    log.info("Updating: `{0}` to `{1}`".format(name, value))
+                    node[name].setValue(str(value))
+
+        # adding closing group knob
+        knob = eval(
+            "nuke.{type}('{name}', '{value}', {group})".format(**knobs[-1]))
+        node.addKnob(knob)
+
+        return node
+
+    except Exception as e:
+        log.warning("Failed to add Avalon data to node: `{}`".format(e))
+        return False
+
+
+def get_avalon_knob_data(node, prefix="ak:"):
+    """ Gets a data from nodes's avalon knob
+
+    Arguments:
+        node (obj): Nuke node to search for data,
+        prefix (str, optional): filtering prefix
+
+    Returns:
+        data (dict)
+    """
+    try:
+        # check if data available on the node
+        test = node['avalon_data'].value()
+        log.debug("Only testing if data avalable: `{}`".format(test))
+    except Exception as e:
```

Comment:
Very broad exception, can this be more specific? E.g. `NameError` or what does it raise?